### PR TITLE
Verify SPIDER_MANAGER_CLASS interface while loading it in CrawlerRunner

### PR DESCRIPTION
**Pull request by Curita**
Discussed in #873.

### CODE REVIEW

**Comment by kmike** [comment]
This change broke some SpiderManager implementations which don't implement `find_by_request` method. `find_by_request` is only useful for `scrapy fetch`, `scrapy parse` and `scrapy shell` commands; crawling should work fine without this method.

File: tests/test_crawler.py [hunk 1]
```diff
@@ -42,3 +43,19 @@ class CustomSettingsSpider(DefaultSpider):
 
         self.assertFalse(settings.frozen)
         self.assertTrue(crawler.settings.frozen)
+
+
+def SpiderManagerWithWrongInterface(object):
```
Comment (kmike) [inline tests/test_crawler.py]: why def?

Comment (Curita) [inline tests/test_crawler.py]: Oh god, why did we merged this (should be 'class', got a little too functional there). I'm not even sure how the test is working.

**Comment by Curita** [comment]
About `find_by_request`, I'm not sure, I kept it in the interface on purpose (It was there before my changes in the Crawler API), I'd like that any SpiderManager implemented all needed functions to run all commands. We're doing this instead of letting it raise an exception later on because some method is not implemented, and that's going to happen eventually if we remove `find_by_request` from the interface (but I grant that another use-case, probably more important, for checking the interface is to verify that the SpiderManager is not using the old interface, and that could be done without `find_by_request` since it's present in the old and the new interface).

I still prefer this check being done over the complete interface, but we could change it.

**Comment by Curita** [comment]
Now that I think of it again maybe we could issue a warning instead of an exception if SpiderManager (soon to be SpiderLoader) doesn't comply with the interface, seems like it's the more tolerant alternative.

Another option could be to raise an exception if the absolutely needed methods to run are not implemented (which I think are `from_settings` and `load`, `list` seems like it's only needed for `scrapy list`) and then issue a warning if the other two methods are not there.

I like both options.

**Comment by MojoJolo** [comment]
@kmike pointed me into this issue. Looks like my problem is related.

I'm having an error and getting ```The find_by_request attribute was not provided.``` message when executing ```scrapy crawl``` or even ```scrapy version```.

**Pull request merged**